src/components/workspace/collabEditing/helper.ts

```diff
@@ -38,3 +38,27 @@ export function createNewSession(onSessionCreated: (sessionId: string) => void)
   xmlhttp.open('GET', 'https://' + LINKS.SHAREDB_SERVER + 'gists/latest/', true);
   xmlhttp.send();
 }
+
+export function checkConnnectionAlive(
+  editorSessionId: string,
+  connectionOK: () => void,
+  sessionIdNotFound: () => void,
+  cannotReachServer: () => void
+) {
+  const xmlhttp = new XMLHttpRequest();
+  xmlhttp.onreadystatechange = () => {
+    if (xmlhttp.readyState === 4 && xmlhttp.status === 200) {

```

Comment:
Avoid magic numbers like 4 and 200. Define them as constants somewhere.

E.g. `const CONNECTION_OK : number = 200;`